File: Modules/ServerTypes/EmployeeManagement/EmployeesColumns.ts
import { ColumnsBase, fieldsProxy } from "@serenity-is/corelib";
import { Column } from "@serenity-is/sleekgrid";
import { EmployeeGender } from "../Modules/EmployeeManagement.Employees.EmployeeGender";
import { EmployeesRow } from "./EmployeesRow";

export interface EmployeesColumns {
    EmployeeId: Column<EmployeesRow>;
    FirstName: Column<EmployeesRow>;
    LastName: Column<EmployeesRow>;
    CertificatesCount: Column<EmployeesRow>;
    Gender: Column<EmployeesRow>;
    BirthDate: Column<EmployeesRow>;
}

export class EmployeesColumns extends ColumnsBase<EmployeesRow> {
    static readonly columnsKey = 'EmployeeManagement.Employees';
    static readonly Fields = fieldsProxy<EmployeesColumns>();
}

[EmployeeGender]; // referenced types

File: Modules/EmployeeManagement/Employees/EmployeesColumns.cs
using Serenity.ComponentModel;
using System;
using System.ComponentModel;
using MySereneApp.Modules.EmployeeManagement.Employees;

namespace MySereneApp.EmployeeManagement.Columns;

[ColumnsScript("EmployeeManagement.Employees")]
[BasedOnRow(typeof(EmployeesRow), CheckNames = true)]
public class EmployeesColumns
{
    [EditLink, DisplayName("Db.Shared.RecordId")]
    public int EmployeeId { get; set; }

    [EditLink, Width(150)] public string FirstName { get; set; }
    [EditLink, Width(150)] public string LastName { get; set; }

    [Width(200)]
    [DisplayName("Forms.Employee.CertificatesCount")]
    public int CertificatesCount { get; set; }

    [Width(100)] public EmployeeGender Gender { get; set; }

    [Width(150), DisplayName("Forms.Employee.BirthDate")]
    public DateTime BirthDate { get; set; }
}